functional-tests/tests/btcio/btcio_read.py

```diff
@@ -31,30 +27,27 @@ def main(self, ctx: flexitest.RunContext):
         # Wait for seq
         wait_for_genesis(seqrpc, timeout=30)
 
-        time.sleep(3)
-
         received_block = btcrpc.getblock(btcrpc.proxy.getbestblockhash())
-        l1stat = seqrpc.strata_l1status()
+        l1stat = wait_until_with_value(
+            lambda: seqrpc.strata_l1status(),
+            lambda value: value["cur_height"] >= received_block["height"],
+            error_with="L1 reader did not catch up with bitcoin network",
+        )
```

Comment:
This should definitely be refactored out into the utils library, as a "wait until the strata node reaches the bitcoin node as of when this function was called" function.